<a href="https://colab.research.google.com/github/YuliiaUtkina/Git_modul_pyhton/blob/main/hw_6_2_%D0%A1%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D1%82%D0%B5%D1%81%D1%82%D0%B8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [20]:
from google.colab import drive
drive.mount('/content/drive')

import pandas as pd
import numpy as np
from scipy import stats

# Зчитування даних з завантаженого файлу
df_july = pd.read_csv('/content/drive/MyDrive/Data/data.csv')

# Запис значень колонки 'data' у форматі np.ndarray
data_july = df_july['data'].to_numpy()


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [21]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [23]:
from statsmodels.stats.weightstats import ztest
# Задання параметрів
mean_check = 20  # середнє значення до липня 2022 року
alpha = 0.05  # рівень значущості
null_mean = 100
z_stat, p_value = ztest(data_july, value=mean_check )

In [24]:
z_stat, p_value

(-0.9953477389335053, 0.31956716690719744)

In [25]:
if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


In [26]:
t_stat, p_value = stats.ttest_1samp(data_july, mean_check )

In [27]:
t_stat, p_value

(-0.9953477389335053, 0.3219937839011988)

In [28]:
if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


**Висновки:**
Результати одновибіркового t-тесту показують:
t-статистика: -0.995
p-значення: 0.322
Оскільки p-значення (0.322) більше рівня значущості (0.05), ми не можемо відхилити нульову гіпотезу. Це означає, що у нас немає статистично значущої різниці у розмірі середнього чеку у липні 2022 року порівняно з попередніми місяцями, де середнє значення було $20.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [29]:
# Зчитування даних з завантаженого файлу
df_august = pd.read_csv('/content/drive/MyDrive/Data/data_2.csv')

# Запис значень колонки 'data' у форматі np.ndarray
data_august = df_august['data'].to_numpy()

In [30]:
mean_july = np.mean(data_july)
mean_august = np.mean(data_august)

In [32]:
print(f"Середній чек у липні 2022 року: {mean_july} доларів")
print(f"Середній чек у серпні 2022 року: {mean_august} доларів")

Середній чек у липні 2022 року: 19.37755319670441 доларів
Середній чек у серпні 2022 року: 19.37322276512519 доларів


In [31]:
z_stat, p_value = ztest(data_july, data_august)

In [33]:
z_stat, p_value

(0.005246673456808504, 0.9958137794593356)

In [34]:
if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


**Висновок:**
Оскільки p-значення (0.996) значно більше рівня значущості (0.05), ми не можемо відхилити нульову гіпотезу. Це означає, що у нас немає статистично значущої різниці у середньому чеку між липнем та серпнем 2022 року.

Висновок з простого спостереження середніх значень також підтверджується статистичним тестом - середні значення чеку у липні та серпні практично однакові, і статистично значуща різниця між ними відсутня.